Restructure docs (#12)
FundingCircle/topology-grapher: README.md

This library provides a means to build a directed graph (data) from a kafka streams topology.

## Rationale

Understanding Kafka topologies can be daunting, and even more so having a high level understanding of the whole system.

With a Kafka-centric architecture it's quite normal to end up with hundreds of topics, with potentially many services consuming them.
So answering questions like "who's consuming this topic", or "who's writing to this topic" can become quite hard, and documentation is very hard to maintain.

> This project is an attempt to simplify developers (and potentially other technical users) lives, giving the ability to generate graphs of Kafka topologies, and generating data in a format that can be consumed to query the generated data.

Reviewer: ```suggestion
This project is an attempt to help developers (and potentially other technical users) understand the Kafka topologies they work with or work on. It generates graphs of Kafka topologies, as data, in a format that can be consumed and queried.
```

> This project alone is intended to be used as a library/script, another component is a UI/API which takes all the data by this library and allows you to display it/query much more easily.

Reviewer: ```suggestion
This project provides a library that, once integrated into an application, enables that application to generate a graph of its Kafka topologies, as data. It can also invoke [GraphViz](http://graphviz.org/) to render the graph to an image file.

Any other usage of the data, such as publishing, viewing, or querying it, is out of scope of this library.
```

> The other component is currently not OSS, but it could be at a later point in time.

Reviewer: ```suggestion
Within Funding Circle we’re also working on a Web app which can read the data produced by this library and allows for viewing and/or querying it via Web browsers and/or anything that can communicate via HTTP. That app is not currently OSS, but it could be at a later point in time.
```

## Status

> This project is being used internally at Funding Circle, but it's still quite *experimental*, so there could be breaking changes.

Reviewer: ```suggestion
This project is being used internally at Funding Circle, but it’s still quite **experimental** so there could be breaking changes.
```

## Tutorial

> (The full code for this tutorial can be found in [sample project](./sample_project/README.md))

Reviewer: ```suggestion
(The full code for this tutorial can be found in [this sample project](./sample_project/README.md).)
```

Suppose you defined a simple topology like:

    (defn transform
      [v]
      (-> v
          (update :value inc)
          (select-keys [:value])))

    (defn t1
      [builder]
      (-> (js/kstream builder (topic-config "input"))
          (js/filter (fn [v] (not (:enabled v))))
          (js/map-values transform)
          (js/to (topic-config "output")))

>       builder)

Reviewer: I suggest using fenced code blocks for the above sample code, and all the code blocks on this page, so that you can specify the language and therefore improve the syntax highlighting that GitHub will use when rendering the code in the page.

Author: Cool done

> And you want to render have a visual idea of how this topology looks like.

Reviewer: ```suggestion
And you want to render a visualisation of this topology.

You could use code like this:
```

     :to-id #uuid "b3f4ee01-7f3d-59f6-84cf-d26f877b6a19"}))}
```

## How to integrate into your application

*The following instructions are just an example of how to automatically publish
the topology edns files to a centralised place, like S3.

> This is specially useful if there is another project reading all these generated

Reviewer: ```suggestion

This is especially useful if there is another project/tool reading all these generated
```

> files, which needs to be built separately.*

Reviewer: ```suggestion
files which needs to be built separately.*
```

> There are a few simple steps to follow to integrate your kafka streaming app.

Reviewer: ```suggestion
There are a few steps to follow to integrate your Kafka Streams app:
```

### 1. Create a namespace or function that invokes `generate-zip`

> ```

Reviewer: ```suggestion
```clojure
```

(ns your-namespace
  (:require [topology-grapher.describe :refer [generate-zip]]))

(def meta-data {:domain "domain"
    :subdomain "subdomain"
    :application "application-name"})

(def topologies
  [{:application-name "app-2"
    :topology topology-1}

   {:application-name "app-1"
    :topology topology-2}])

(generate-zip topologies meta-data)

> 

Reviewer: ```suggestion
```

```

The `generate-zip` function is what takes care of transforming
topologies objects into EDN files that are written out to disk.

The important thing is that you need to be have a function that
returns a topology object *without* having to actually start Kafka.

> This is normally fine, but if you are using something like the Stuart

Reviewer: ```suggestion
This is normally fine, but if you are using something like Stuart
```

> Sierra [component](https://github.com/stuartsierra/component) for example you will need to refactor the

Reviewer: ```suggestion
Sierra’s [component](https://github.com/stuartsierra/component) for example you may need to refactor the
```
